op-chain-ops/genesis/layer_two.go

```diff
@@ -46,10 +46,10 @@ func BuildL2Genesis(config *DeployConfig, dump *foundry.ForgeAllocs, l1StartBloc
 	}
 	genspec.Alloc = dump.Copy().Accounts
 	// ensure the dev accounts are not funded unintentionally
-	if hasDevAccounts, err := HasAnyDevAccounts(genspec.Alloc); err != nil {
+	if devAccounts, err := RetrieveDevAccounts(genspec.Alloc); err != nil {
 		return nil, fmt.Errorf("failed to check dev accounts: %w", err)
-	} else if hasDevAccounts != config.FundDevAccounts {
-		return nil, fmt.Errorf("deploy config mismatch with allocs. Deploy config fundDevAccounts: %v, actual allocs: %v", config.FundDevAccounts, hasDevAccounts)
+	} else if len(devAccounts) > 0 && !config.FundDevAccounts {
```

Comment:
```suggestion
	} else if (len(devAccounts) > 0) != config.FundDevAccounts {
```
To preserve original logic.
If `FundDevAccounts` is true, then there should be dev accounts. If false, then there should be none. Other combinations are misconfigurations.